src/faction_camp.cpp

```diff
@@ -1683,12 +1714,11 @@ void basecamp::start_fortifications( std::string &bldg_exp )
                                       _( "begins constructing fortifications..." ), false, {},
                                       making.skill_used.str(), making.difficulty );
         if( comp != nullptr ) {
-            g->u.consume_components_for_craft( making, fortify_om.size() * 2 - 2, true );
-            g->u.invalidate_crafting_inventory();
-            comp->companion_mission_role_id = bldg_exp;
-            for( auto pt : fortify_om ) {
-                comp->companion_mission_points.push_back( pt );
-            }
+            consume_components_for_camp_mission( omt_tri, making, fortify_om.size() * 2 - 2, by_radio );
+        }
+        comp->companion_mission_role_id = bldg_exp;
```

Comment:
I guess whole block below should go to under `if( comp != nullptr )` check. Right now it could crash if `comp == nullptr` and it would try to assign `companion_mission_role_id` or fill-in `companion_mission_points`.